### ISSUE TITLE
Compilation issue with mingw-64 and USE_IPV6

### ISSUE THREAD

**Issue by etrinh** [bel2125/civetweb#57]
Hi,

Changing the ifdef condition to enable the redefinition of "struct pollfd" makes the work.

Best Regards,
Eric

**Comment by bel2125** [comment]
I don't have a MinGW-64 installed at the moment - I will have to do this later, but for the moment please give me some more information:

You mean this definition?
https://github.com/bel2125/civetweb/blob/master/src/civetweb.c#L254

    #if !defined(USE_IPV6) && defined(_WIN32)
    #ifndef HAVE_POLL
    struct pollfd {
        ...

In your case, USE_IPV6 is defined, _WIN32 should be defined (is it?). 
Is HAVE_POLL defined?

It seems like mingw64 uses other windows header as Visual Studio.

**Comment by etrinh** [comment]
Hi,
Yes, it is exactly this line (254).
I use Mingw64 (gcc 4.8.0)  (mingw API is version 3)
The HAVE_POLL is not defined as well as the struct pollfd.

I just commented the condition USE_IPV6 to compile correctly, in my case, but maybe there are several checks to take care.

Thanks,
Eric

**Comment by bel2125** [comment]
I installed mingw-w64 on Linux, and tried to compile the current source.
I had no problems with poll, but `char *_getcwd(char *buf, size_t size);` in main.c did not compile. Fixing this, I had a long, long list of link errors, so a MinGW64 setup does not seem so easy, and I probably won't find the time to reproduce this soon.
I'm not sure if I break another build by just changing the conditions.

**Comment by bel2125** [comment]
I have tried for several hours to get mingw running om Linux and Windows - without success.
I also did not find complete installation instructions, but only here and there a snippet.
I had problems installing a proper make, problems with installing the compiler, problems with system include files, now I have problems with the linker ... in particular with winsockets ... sorry, but I can not spent any more time on trying to get make running with mingw.

Also, I was not able to reproduce the compile error you mentioned above.
Maybe it is not related to the compiler but to the SDK.
